packages/app/src/cli/services/app/config/push.ts

```diff
@@ -105,44 +135,35 @@ export async function pushConfig(options: PushOptions) {
   })
 }
 
-const getMutationVars = (app: App, configuration: CurrentAppConfiguration) => {
-  let webhookApiVersion
-  let gdprWebhooks
-
-  if (app.betas?.declarativeWebhooks) {
-    // These fields will be updated by the deploy command
-    webhookApiVersion = app.webhookApiVersion
-    gdprWebhooks = app.gdprWebhooks
-  } else {
-    webhookApiVersion = configuration.webhooks?.api_version
-    gdprWebhooks = {
-      customerDeletionUrl: configuration.webhooks?.privacy_compliance?.customer_deletion_url,
-      customerDataRequestUrl: configuration.webhooks?.privacy_compliance?.customer_data_request_url,
-      shopDeletionUrl: configuration.webhooks?.privacy_compliance?.shop_deletion_url,
-    }
-  }
+const getMutationVars = (app: App, localApp: AppInterface) => {
+  const configuration = localApp.configuration as CurrentAppConfiguration
 
+  const appHomeSchema = localApp.getConfigExtension(AppHomeSpecIdentifier) as AppHomeConfiguration
+  const posSchema = localApp.getConfigExtension(PosSpecIdentifier) as PosConfiguration
```

Comment:
I don't think we should access the config like this.
If the app has loaded all the config, why not merge everything in the config object and just use `localApp.configuration...`? Since you have a unified schema that configuration can be typed and will be better than casting everytime no?